sse/src/test/scala/akka/stream/alpakka/sse/EventSourceSpec.scala

```diff
@@ -0,0 +1,156 @@
+    private case object Unbind
+
+    def route(setEventId: Boolean): Route = {
+      import Directives._
+      import EventStreamMarshalling._
+      get {
+        optionalHeaderValueByName(`Last-Event-ID`.name) { lastEventId =>
+          try {
+            val fromSeqNo = lastEventId.getOrElse("0").trim.toInt + 1
+            complete {
+              Source.fromIterator(() => Iterator.from(fromSeqNo).take(2)).map(toServerSentEvent(setEventId))
+            }
+          } catch {
+            case _: NumberFormatException =>
+              complete(
+                HttpResponse(
+                  BadRequest,
+                  entity = HttpEntity(
+                    `text/event-stream`,
+                    "Integral number expected for Last-Event-ID header!".getBytes(UTF_8)
+                  )
+                )
+              )
+          }
+        }
+      }
+    }
+  }
+
+  class Server(address: String, port: Int, route: Route) extends Actor with ActorLogging {
+    import Server._
+    import context.dispatcher
+
+    private implicit val mat = ActorMaterializer()
+
+    self ! Bind
+
+    override def receive = unbound
+
+    private def unbound: Receive = {
+      case Bind =>
+        Http(context.system).bindAndHandle(route, address, port).pipeTo(self)
+        context.become(binding)
+    }
+
+    private def binding: Receive = {
+      case serverBinding @ Http.ServerBinding(socketAddress) =>
+        log.info("Listening on {}", socketAddress)
+        context.system.scheduler.scheduleOnce(1500.milliseconds, self, Unbind)
+        context.become(bound(serverBinding))
+
+      case Status.Failure(cause) =>
+        log.error(cause, s"Can't bind to {}:{}!", address, port)
+        context.stop(self)
+    }
+
+    private def bound(serverBinding: Http.ServerBinding): Receive = {
+      case Unbind =>
+        serverBinding.unbind().map(_ => Done).pipeTo(self)
+        context.become(unbinding(serverBinding.localAddress))
+    }
+
+    private def unbinding(socketAddress: InetSocketAddress): Receive = {
+      case Done =>
+        log.info("Stopped listening on {}", socketAddress)
+        context.system.scheduler.scheduleOnce(500.milliseconds, self, Bind)
+        context.become(unbound)
+
+      case Status.Failure(cause) =>
+        log.error(cause, s"Can't unbind from {}!", socketAddress)
+        context.stop(self)
+    }
+  }
+
+  private def toServerSentEvent(setEventId: Boolean)(n: Int) = {
+    val id = n.toString
+    val event = ServerSentEvent(Some(id))
+    if (setEventId) event.copy(id = Some(id)) else event
+  }
+}
+
+class EventSourceSpec extends AsyncWordSpec with Matchers with BeforeAndAfterAll {
+  import EventSourceSpec._
+
+  private implicit val system = ActorSystem()
+  private implicit val ec = system.dispatcher
+  private implicit val mat = ActorMaterializer()
+
+  "EventSource" should {
+    "communicate correctly with an instable HTTP server" in {
+      val host = "localhost"
+      val port = 9999
```

Comment:
@johanandren I can't find that, but there's `akka.util.SocketUtil#temporaryServerAddress`.